### ISSUE TITLE
Grid cannot post new data to server when pressing "Add new" button in toolbar. Is it an issue?

### ISSUE THREAD

**Issue by ralf-dzq** [vitmalina/w2ui#968]
I know I could use $.ajax() to do it by myself in onAdd().

However my question is **why a grid with "crud" can auto-post info to server when updating, deleting, reading, but not when adding(creating)**.

Or maybe my code is wrong?

``` javascript
<!DOCTYPE html>
<html>
<head>
    <title>W2UI Demo: grid-1</title>
    <link rel="stylesheet" type="text/css" href="http://w2ui.com/src/w2ui-1.4.2.min.css" />
    <script src="http://ajax.googleapis.com/ajax/libs/jquery/2.1.0/jquery.min.js"></script>
    <script type="text/javascript" src="http://w2ui.com/src/w2ui-1.4.2.min.js"></script>
</head>
<body>

<div id="grid" style="width: 100%; height: 350px;"></div>

<script type="text/javascript">
$(function () {
    $('#grid').w2grid({ 
        name: 'grid', 
        url  : {
            get    : '/data/get',
            remove : '/data/remove',
            save   : '/data/save'
        },
        method: 'GET', // need this to avoid 412 error on Safari
        show: { 
            toolbar: true,
            footer: true,
            toolbarAdd: true,
            toolbarSave: true
        },
        columns: [                
            { field: 'fname', caption: 'First Name', size: '30%' },
            { field: 'lname', caption: 'Last Name', size: '30%' },
            { field: 'email', caption: 'Email', size: '40%' },
            { field: 'sdate', caption: 'Start Date', size: '120px' }
        ],
        onAdd: function(event) {
            var g = w2ui['grid'].records.length;
            w2ui['grid'].add( { recid: g + 1, fname: 'Jin', lname: 'Franson', email: 'jdoe@gmail.com', sdate: '4/3/2012' } );  
        }
    });    
});
</script>

</body>
</html>

```

**Comment by vitmalina** [comment]
The indention of onAdd is not to submit it to the server, but to provide you a hook how to submit. I personally is not a fan of inline editing. It is great for some things (excel like grids), but it is not a silver bullet for all, and people seem to use it too much (IMHO). However, you can use w2form, which will submit your record to the server as well as give you other functionality. It cannot be inline, it is either in a popup, or in a div next to the list (or instead the grid). I hope I have answered your question.

Since it is not a bug, I am closing this issue.

**Comment by hansgru** [comment]
> I personally is not a fan of inline editing. It is great for some things (excel like grids), but it is not a silver bullet for all, and people seem to use it too much (IMHO). 

Most business applications require it because it's very fast to do things (if also works smoothly with keyboard only) . Not all the data needs to be edited like this, but in many scenarios, the inline mode is faster and more secure, because the user can do many changes and save only at once when everything fits.

Inline edit with w2ui would have the advantage over http://handsontable.com/ :
 
  - it can behave like Excel, but keep IDs in background in the data model (so like excel but with "foreign keys")
  - it can highlight what's changed so that the user decides later to save all at once (handsontable is using the red corner for comments not changes)

IMHO it would be very important for w2ui to work easily in inline mode, and allow multi-selects and custom field types too.

**Comment by vitmalina** [comment]
I agree with many your points. I am currently working on a project where w2grid is used as spreadsheet and am adding changes to make it smoother. Thanks for your feedback.
